src/darwin/Framework/CHIP/MTRDeviceControllerStartupParams.h

```diff
@@ -14,8 +14,6 @@
  *    limitations under the License.
  */
 
-#import <Foundation/Foundation.h>
```

Comment:
Again, we are using NSObject here.  We should be importing the right header for that, not randomly getting it from something else we import.

Reply:
We have dozens of headers that don't explicitly import Foundation, generally they get it via `MTRDefines.h`. And any header that exports anything will be using `MTR_EXPORT` or an availability macro from MTRDefines anyway, so to me that seems like an acceptable way to reduce boilerplate.

You seem fine with relying on Foundation to import import `dispatch.h` for us, how is this different?

Reply:
> You seem fine with relying on Foundation to import dispatch.h for us, how is this different?

I did not realize dispatch.h is a separate thing.  ;)

But more to the point, I assume that Foundation importing dispatch.h is part of its API contract, right?  We don't really have a clear API contract for what MTRDefines.h imports.

If we're OK saying it always imports Foundation and we depend on that, I'm OK with that.